### PR TITLE
Fix nodeport allocation race condition

### LINKED ISSUE #54885: Service Nodeport and Healthcheck Nodeport Collision
I came across a cluster where it has 2 services with port collision. 

One service looks like this:
```
type: LoadBalancer
externalTrafficPolicy: Local
healthCheckNodePort: ${Collided Port}
```

The other looks like:
```
...
  annotation: cloud.google.com/load-balancer-type=Internal
Spec: 
  type: LoadBalancer
  externalTrafficPolicy: Cluster
  ports
  - nodeport: ${Collided Port}
    ....
```

I checked the logic that allocates nodeport and healthcheck nodeport. They went thru the same node port allocator. Port allocation mainly happens in `initNodePorts`, `updateNodePorts` and `allocateHealthCheckNodePort`. The file in question is https://github.com/kubernetes/kubernetes/blob/release-1.8/pkg/registry/core/service/rest.go

### PR DESCRIPTION
**Pull request by xiangpengzhao**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
~~Fixes #54885~~

**Special notes for your reviewer**:
/cc @freehan 

**Release note**:

```release-note
none
```

### CODE REVIEW

File: pkg/registry/core/service/portallocator/operation.go [hunk 1]
```diff
@@ -95,6 +100,8 @@ func (op *PortAllocationOperation) Commit() []error {
 
 // Allocates a port, and record it for future rollback
 func (op *PortAllocationOperation) Allocate(port int) error {
```
Comment (smarterclayton) [inline pkg/registry/core/service/portallocator/operation.go]: Is `op` called in a threaded context?  If so, this change would be correct.  Allocate *should* be thread safe and only allocate a port once, but double checking that would be good.

Comment (freehan) [reply pkg/registry/core/service/portallocator/operation.go]: I do not think so. It is called in https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/service/rest.go

**Comment by freehan** [comment]
The bitmap is already mutex protected. No need another one in the wrapper
https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/service/allocator/bitmap.go

**Pull request closed without merging**